### DISCUSSION TITLE
ioBroker hm-rpc aktualisiert nicht alle Werte im Objektbaum

### DISCUSSION THREAD

**Discussion by jogicom** [OpenCCU/OpenCCU#2355] [🇩🇪 Fragen&Antworten] [answered]
Hallo, ich habe diesbezüglich auch schon im ioBroker Forum nachgefragt, dort heißt es, die RaspberryMatic sei schuld und rückt die Werte nicht raus. Leider kenne ich mich mit der Kommunikation rpc-Adapter<--->Homematic nicht aus und stehe deshalb etwas im Regen. Vielleicht kann ja hier jemand etwas Licht ins Dunkle bringen.....

Ich habe eine RaspberryMatic (ELV-Charly) in aktueller Version, daran hängt ein HomematicGateway, das per Netzwerk mit der RaspberryMatic verbunden ist, das Gateway ist erreichbar und funktioniert.

Nach einem Netzwerkausfall, gingen die an dem Gateway angelernten Geräte auf Unreach und wurden dann in der HM auch als gestört gemeldet. Nach Reparatur der Netzwerkverbindung werden in der Homematic alle Geräte wieder als erreichbar gemeldet.

Nun habe ich das Problem, dass nicht alle Werte im Objektbaum des hm-rpc Adapters aktualisiert werden. Anhand der Zeitstempel wurden die Werte das letzte mal vor einem halben Jahr aktualisiert, wo ich vermutlich ioBroker neu aufgesetzt habe.

Mir fällt auf, dass vor allem die Daten im Objektbaum der Geräte im Index 0 schon lange nicht mehr aktualisiert wurden, zum Beispiel die RSSI Werte, Unreach usw. trägt alles den Zeitstempel von vor einem halben Jahr.

Wenn ich die Geräte in der hm-rpc Instanz neu einlese, bleiben die Zeitstempel auf den alten Werten.

Jetzt könnte man ja sagen, dass das Problem an der HM liegt, jedoch ist es so, dass wenn ich den Adapter deinstalliere und neu installiere, die richtigen Werte in den Zuständen drin sind. Das hat natürlich den großen Nachteil, dass ich dann das SQL Logging wieder für jeden Parameter konfigurieren muss, was eine Katastrophe ist.

Wie oben schon erwähnt, soll die RaspberryMatic die Werte nicht rausrücken, was ich mangels Kenntnis der Funktionsweise nicht nachvollziehen kann.

Wenn ich mit der XML-API die States eines Gerätes aus der Zentrale auslese, sieht es folgendermaßen aus:
```
<state>'

<device name="Gartenpumpe-Keller" ise_id="*hidden*" unreach="false" sticky_unreach="false" config_pending="false">

<channel name="Gartenpumpe-Keller:0" ise_id="*hidden*">

<datapoint name="BidCos-RF.*hidden*:0.UNREACH" type="UNREACH" ise_id="26345" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.STICKY_UNREACH" type="STICKY_UNREACH" ise_id="26341" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.CONFIG_PENDING" type="CONFIG_PENDING" ise_id="26327" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.DUTYCYCLE" type="DUTYCYCLE" ise_id="26335" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.RSSI_DEVICE" type="RSSI_DEVICE" ise_id="26339" value="-70" valuetype="16" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.RSSI_PEER" type="RSSI_PEER" ise_id="26340" value="-62" valuetype="16" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.DEVICE_IN_BOOTLOADER" type="DEVICE_IN_BOOTLOADER" ise_id="26331" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>

<datapoint name="BidCos-RF.*hidden*:0.UPDATE_PENDING" type="UPDATE_PENDING" ise_id="26349" value="false" valuetype="2" valueunit="" timestamp="1688731914"/>
```

Wobei der Timestamp der Zeit Fri Jul 07 2023 14:11:54 GMT+0200 entspricht.

Schaue ich mir die Daten in IoBroker an:
![grafik](https://github.com/jens-maus/RaspberryMatic/assets/25620021/ac6bf546-ead9-40e5-9f44-e1835eaecc97)

Und hier der Unreach Zeitstempel von ioBroker:
![grafik](https://github.com/jens-maus/RaspberryMatic/assets/25620021/e6b17d33-6a86-44d6-9506-1dca253e6e8c)
Das war der Zeitpunkt des Netzwerkausfalls.

Ich schaffe es nicht, die Daten wieder Synchron zu bekommen, weil angeblich die HM die Daten nicht rausrückt. Einzige möglichkeit wäre, die entsprechende Instanz in IOBroker zu löschen und dann wieder anzulegen, was jedoch weitreichende Konsequenzen hat.

Vielleicht kann mich hier jemand, mit dem nötigen Hintergrundwissen, etwas aufklären?

Edit: [Hier der Link zum ioBroker Formum](https://forum.iobroker.net/topic/66746/hm-rpc-aktualisiert-nicht-alle-werte-im-objektbaum?_=1688830775701)

**Comment by jp112sdl** [comment]
Hilft es, wenn du in den Instanz-Einstellungen den Haken setzt (+ "Speichern und schließen")?
<img width="325" alt="Bildschirmfoto 2023-07-09 um 10 24 52" src="https://github.com/jens-maus/RaspberryMatic/assets/18190139/02b6d81c-63ad-4492-b8ee-9952f7826007">

**Reply by jogicom** [reply]
Schon mehrfach versucht, hilft leider nichts.

**Comment by jens-maus** [comment]
> Hallo, ich habe diesbezüglich auch schon im ioBroker Forum nachgefragt, dort heißt es, die RaspberryMatic sei schuld und rückt die Werte nicht raus.

Link? Wo wird das genau thematisiert im ioBroker Forum?

**Reply by jogicom** [reply]
[Link zum ioBroker Forum Thema](https://forum.iobroker.net/topic/66746/hm-rpc-aktualisiert-nicht-alle-werte-im-objektbaum?_=1688830775701)

**Reply by jens-maus** [reply]
Da steht aber nix davon das explizit RaspberryMatic dran schuld ist, sondern hier wird immer von Homematic gesprochen. Der Punkt ist einfach, das dss was du via xmlapi da dir ausgeben lässt eben wa anderes ist was via xmlrpc event-basiert an hm-rpc geliefert wird. Deshalb lässt sich das zwangsläufig nicht immer 1:1 vergleichen.

Auch verstehe ich immer noch nicht wo hier konkret das Problem bzw die Inkosistenz ist, zumindest sehe ich sie nicht, ausser die Zeitstempel natürlich. Aber die sind wurscht im Grunde. Also bitte noch einmal detaillierter anhand eines Gerätes, eines Parameters reproduzierbar die Differenz darstellen und dazu sagen welcher nun der "Wahrheit" entspricht. Nur so lässt sich dem vermeintlichen Problem suf den Grund gehen - wenn es denn ein prinzipielles Problem ist. Aktuell glaube ich aber eben mehr daran das das ggf ein sideeffect des Weges wie der hm-rpc adapter arbeitet ist.

Und abgesehen davon, ich setze selbst ioBroker und hm-rpc breit ein und hab noch nie solche Inkonsistenzen bei Objekten/Parametern festgestellt...

**Reply by jogicom** [reply]
Das Problem habe ich extrem selten, und NUR wenn es einen Netzausfall zwischen der RaspberryMatic und dem Gateway gab.

Wenn ich mir die Werte in ioBroker ansehe, ist dort der **State UREACH auf true**, wenn ich den State aus der RaspBerryMatic auslese ist der **State false**.

Dass im ioBroker Forum die Homematic & RasperyMatic verwuschtelt werden, da kann ich nun wirklich nichts dafür, im einleitenden Post habe ich ja ausdrücklich die RaspberryMatic angegeben.

Ich werde jetzt mal für ein paar Stunden einige Geräte, die in ioBroker als UREACH signalisiert werden, abschalten und warten bis in der RaspberryMatic die Störungen anstehen, mal sehen ob sich das dann wieder "normalisiert"

**Comment by jogicom** [comment]
Obwohl es daran eigentlich nicht liegen kann, hier mal noch die einstellungen der RaspberryMatic Firewall, man weis ja nie...

![grafik](https://github.com/jens-maus/RaspberryMatic/assets/25620021/080fb739-c0a0-4cac-9a23-28d4e63b6a23)

Die Ports im Detail:
```
2000;
2001;
2002;
2010;
3493;
8701;
9292;
32001;
32010;
42000;
42001;
42010;
49292
```

**Reply by jens-maus** [reply]
> Obwohl es daran eigentlich nicht liegen kann, hier mal noch die einstellungen der RaspberryMatic Firewall

Na wie soll es denn bitte an der Firewall liegen das manche Objekte aktuell sind und manche nicht? Entweder ist die firewall zu, dann wird dir iobroker einen fehler bringen oder nicht, dann geht die XMLRPC Kommunikation.

Ich würd vorschlagen du baust dir mal ne Testumgebung einer frischen CCU/RaspberryMatic mit nur einem Gerät und demonstrierst hier die vermeintliche Inkonsistenz von grund auf neu wenn diese in der frischen Ungebung auch auftritt. Wie schon angedeutet: bis jetzt sieht das für mich nicht nach einem problem aus.

**Reply by jogicom** [reply]
Bezüglich Firewall: Ich weiß, aber irgendwann greift man nach jedem Strohhalm :-)

**Comment by jogicom** [comment]
Die gute Nachricht: Nachdem ich zum testen einige Geräte stromlos gemacht habe und die Störungsmeldungen in der RM kamen, werden die States wieder aktualisiert.

Was mir aber in ioBroker aufgefallen ist :
Der **State UREACH** ist auf **TRUE**
Der **State UNREACH-ALARM** steht auf **No Alarm(0)**

**Also Ursprungszustand in ioBroker:**
UNREACH  = true
UNREACH-ARLAM = NoAlarm(0)

**Gerät wird stromlos und RM registriert Kommunikationsstörung:**
UNREACH = true
UNREACH-ALARM = Alarm(1)

**Geräte wieder angeschaltet:**
UNREACH = false
UNREACH-ALARM = NoAlarm(0)


Bei einem tatsächlich ausgefallenem Gerät sollte UNREACH-ALARM auf **Alarm(1)** sein.
Soweit ich das beurteilen kann kommt der state UNREACH-ALARM nicht von der RM, zumindest finde ich dort den State nicht. Liege ich richtig, dass der state UNREACH-ALARM von ioBroker selbst generiert wird ?

**Comment by jogicom** [comment, marked as answer]
Also, das einzige was wirklich hilft, wenn sich die States zwischen der Zentrale und ioBroker verhakt haben:

1. Die Geräte stromlos machen
2. Unreach in der Zentrale abwarten
3. Geräte wieder in Betrieb nehmen

Danach sind die States in ioBroker und RaspberryMatic wieder synchron.

Danke für euren Input!
